stix2/utils.py

```diff
@@ -29,32 +48,47 @@ def format_datetime(dttm):
     else:
         zoned = dttm.astimezone(pytz.utc)
     ts = zoned.strftime("%Y-%m-%dT%H:%M:%S")
-    if zoned.microsecond > 0:
-        ms = zoned.strftime("%f")
+    ms = zoned.strftime("%f")
+    precision = getattr(dttm, "precision", None)
+    if precision:
+        if precision == "millisecond":
+            ts = ts + '.' + ms[:3]
```

Comment:
I don't think we need both nested `if`s. In fact, we can probably replace lines 52-54 with `if getattr(dttm, "precision", None) == "millisecond":`